lib/universal/models/attachment.rb

```diff
@@ -24,6 +25,25 @@ module Attachment
         scope :for_comment, -> (c){where(tci: c)}
         scope :for_name, ->(n){where(name: n)}
         scope :recent, ->(){order_by(created_at: :desc)}
+        before_update :download_uploaded_file
+        after_update :move_uploaded_file
+        
+        def download_uploaded_file
+          if !self.temporary_comment_id.blank?
+            new_file_path = Rails.root.join('tmp', "#{self.file_filename}")
+            File.open(new_file_path, 'wb') do |f|
+              f << open("http:#{self.file.url.gsub(/hms\/(.*)attachment/, "hms/attachment")}").read
+            end
+          end
+        end
+        
+        def move_uploaded_file
+          if !self.temporary_comment_id.blank?
```

Comment:
I think you will need to check if the subject has been updated.
`if this.subject_id_changed?`

Also, you should set the `temporary_comment_id` to nil after you have moved.
`set.unset(:temporary_comment_id)`

Reply:
I'm creating an entirely new attachment on line 43, then deleting the existing one on line 44, so it shouldn't be necessary to unset the value right? 

Is there a more efficient way to "move" the original instead of creating a new attachment?

Reply:
Sorry, you're right.